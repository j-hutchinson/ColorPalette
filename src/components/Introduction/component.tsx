import React from 'react';
import styled from 'styled-components';
import space from './space.jpg'
import { SectionStyles } from '../../css/tokens';

const StyledWrapper = styled.section`
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 48px;
    ${SectionStyles};

    @media screen and (max-width: 800px) {
        grid-template-columns: 1fr;
        grid-template-rows 0.3fr 0.7fr;
    }
`;

const StyledText = styled.div`
    margin: auto;
`;

const StyledHeading = styled.h1`
    font-size: 80px;
    line-height: 64px;
    color: white;
    margin-bottom: 16px;
    margin-top: 0px;

    @media screen and (max-width: 800px) {
        font-size: 48px;
    }
`;

const StyledPara = styled.p`
    font-size: 24px;
    line-height: 36px;
    color: white;
    margin: 0px;

    @media screen and (max-width: 800px) {
        font-size: 20px;
    }
`;

const StyledImg = styled.img`
    margin: auto;

    @media screen and (max-width: 800px) {
        width: 180px;
        margin: 24px 20px;
    }
`;

const StyledLink = styled.a`
    color: white;
    font-style: italic;
    font-weight: 500;  
    text-decoration: none;

    :hover {
        color: linear-gradient(to right, white 34%, blue 65%);
        transition: all 0.5s;
    }
`;

const Introduction = (): JSX.Element => (
    <StyledWrapper>
        <StyledText>
            <StyledHeading>Hi, I'm Jack ðŸ‘‹</StyledHeading>
            <StyledPara>I'm a full stack web developer at{' '}
                <StyledLink href="https://checkout.com" target="_blank" rel="noopener noreferrer">
                    Checkout.com
                </StyledLink>
            </StyledPara>
        </StyledText>
        <StyledImg src={space} width="380px" />
    </StyledWrapper >
);

export default Introduction;
