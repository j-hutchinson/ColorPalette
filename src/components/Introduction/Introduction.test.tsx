import { shallow } from 'enzyme';
import React from 'react';
import Introduction from './component';

describe('Introduction component', () => {
    test('component matches snapshot', () => {
        expect.assertions(1);

        const wrapper = shallow(<Introduction />);

        expect(wrapper).toMatchSnapshot();
    });
});