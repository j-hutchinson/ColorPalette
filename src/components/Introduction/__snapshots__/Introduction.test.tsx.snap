// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Introduction component component matches snapshot 1`] = `
<styled.section>
  <styled.div>
    <styled.div>
      <styled.h1>
        Jack Hutchinson
      </styled.h1>
      <styled.h2>
        Full stack engineer at
         
        <styled.a
          href="https://checkout.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          Checkout.com
        </styled.a>
      </styled.h2>
      <styled.p>
        Building reliable and scalable systems by day. Leveraging technology to launch fully featured products by night.
      </styled.p>
      <Socials />
    </styled.div>
    <styled.img
      src="space.jpg"
    />
  </styled.div>
</styled.section>
`;
